notebooks/use-cases/information-extraction.ipynb

```diff
@@ -0,0 +1,285 @@
+   "outputs": [],
+   "source": [
+    "from docling.datamodel.base_models import InputFormat\n",
+    "from docling.document_extractor import DocumentExtractor\n",
+    "\n",
+    "extractor = DocumentExtractor(allowed_formats=[InputFormat.IMAGE, InputFormat.PDF])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "extraction-templates",
+   "metadata": {},
+   "source": [
+    "## Information Extraction with Templates\n",
+    "\n",
+    "Docling supports different template formats for information extraction. Templates define the structure and data types of the information you want to extract from documents. Let's explore the different approaches:\n",
+    "\n",
+    "### String Template Format\n",
+    "\n",
+    "Templates can be string literals in JSON format. This is the simplest approach for basic extraction needs:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "template-definition",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "res = extractor.extract(invoice_url, template='{\"invoice_number\": \"string\", \"total\": \"float\"}')\n",
+    "print(res)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "vlm-pipeline",
+   "metadata": {},
+   "source": [
+    "### Dictionary Template Format\n",
+    "\n",
+    "Templates can also be Python dictionaries, which provides better integration with Python code and allows for more complex data structures:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "vlm-setup",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "res = extractor.extract(invoice_url, template={\"invoice_number\": \"string\", \"total\": \"float\"})\n",
+    "print(res)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "extraction-demo",
+   "metadata": {},
+   "source": [
+    "### Pydantic Model Template Format\n",
+    "\n",
+    "For more advanced use cases, templates can be [Pydantic](https://docs.pydantic.dev/latest/) model classes or instances. This approach provides type validation, default values, and better documentation:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "run-extraction",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import Optional\n",
+    "\n",
+    "class Invoice(BaseModel):\n",
+    "    invoice_number: str = Field(examples=[\"INV-001\", \"12345\"])\n",
+    "    total: float = Field(default=10, examples=[100.0, 250.50])\n",
+    "    vendor_name: Optional[str] = Field(default=None, examples=[\"ACME Corp\", \"Tech Solutions Inc\"])\n",
+    "\n",
+    "res = extractor.extract(invoice_url, template=Invoice)\n",
+    "print(res)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "evaluation",
+   "metadata": {},
+   "source": [
+    "### Pydantic Instance Template Format\n",
+    "\n",
+    "You can also use a Pydantic model instance as a template, which allows you to override the defaults and provide specific fallback values:"
```

Comment:
Only after reading the [docling example for this](https://docling-project.github.io/docling/examples/extraction/#using-a-pydantic-model-template) did I understand why you added this section. 

Can you add something grounded in the invoice we're using in the notebook similar to this snippet in the notebook had when they mentioned:
```
This can be very useful in scenarios where we happen to have available context that is more relevant than the default values predefined in the model definition.

E.g. in the example below:

    bill_no and total are actually set from the value extracted from the data,
    there was no tax_id to be extracted, so the updated default we provided was applied
```